test/e2e/tests/test_external_oidc.py

```
@@ -869,39 +869,37 @@ def test_injected_subscription_header_ignored(self, maas_api_base_url: str):
-    def test_injected_username_on_oidc_token_ignored(self, maas_api_base_url: str):
-        """Client-supplied X-MaaS-Username with a raw OIDC token (not API key) is ignored.
+    def test_injected_username_on_oidc_token_rejected(self, maas_api_base_url: str):
+        """Client-supplied X-MaaS-Username with a raw OIDC token is rejected on mint.
 
         Use a raw OIDC bearer token (not an API key) and inject a spoofed
-        username. The minted API key should still reflect the real OIDC
-        identity, not the injected header.
+        username. deny-client-identity-headers must deny before maas-api mints
+        a key, so no key material is returned.
         """
         token = _request_oidc_token(username="alice_lead", password="letmein")
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to argument: "password"

(S106)